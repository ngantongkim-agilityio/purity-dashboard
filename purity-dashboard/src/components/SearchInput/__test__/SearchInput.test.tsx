import { render } from '@testing-library/react';

// Components
import { SearchInput } from '..';

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn().mockReturnValue({
    get: jest.fn().mockReturnValue('test')
  }),
  useRouter: jest.fn().mockReturnValue({
    replace: jest.fn()
  }),
  usePathname: jest.fn().mockReturnValue('')
}));

describe('SearchInput component', () => {
  it('Should render SearchInput component with default props', () => {
    const { container } = render(<SearchInput />);

    expect(container).toMatchSnapshot();
  });
});
