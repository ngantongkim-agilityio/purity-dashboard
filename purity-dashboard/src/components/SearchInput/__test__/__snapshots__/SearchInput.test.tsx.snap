// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component Should render SearchInput component with default props 1`] = `
<div>
  <div
    class="relative flex flex-1 flex-shrink-0"
  >
    <label
      class="sr-only"
      for="search"
    >
      Search
    </label>
    <input
      class="peer block w-full rounded-md border border-gray-200 py-[9px] pl-10 text-sm outline-2 placeholder:text-gray-500"
      placeholder="Search..."
      value="test"
    />
    <svg
      aria-hidden="true"
      class="absolute left-3 top-1/2 h-[18px] w-[18px] -translate-y-1/2 text-gray-500 peer-focus:text-gray-900"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </div>
</div>
`;
